Basic options contract and test file

TobiasBK opened the pull request:
Basic options contract that allows for covered calls and puts. The underlying is ETH and the payment is in DAI. This is a work in progress. I wanted to share what I had so far. I haven't written any test yet, will do in the next few days. Let me know what you think.

PatrickAlphaC commented:
This is so cool!!!!!

Yes, would love to see some tests and would be happy to merge. The use of the price feeds to get the conversion for the options is awesome!!!

TobiasBK commented:
I've added deploy script. Updated contract comments and fixed some basic errors. I also wrote some initial tests. A lot more tests can be written, but I won't be able to get to it until next weekend. 

Hopefully this contract is interesting enough to get others to make it better. :) 

Let me know if you want anything changed, see any mistakes (there are probably a few), etc.

PatrickAlphaC commented:
Looks good!!

I'll give it a deeper dive in the coming month, we'll leave it as in progress for the moment.
